Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -604,23 +605,35 @@ void WiiSocket::DoSock(Request request, SSL_IOCTL type)
   pending_sockops.push_back(so);
 }
 
-void WiiSockMan::AddSocket(s32 fd)
+s32 WiiSockMan::AddSocket(s32 fd)
 {
   if (fd >= 0)
   {
-    WiiSocket& sock = WiiSockets[fd];
+    s32 wii_fd = 3;
```

Comment:
No idea. That's why it needs a hardware test to know at least the following:
 - the max_fd value (if any)
 - the maximum number of open sockets the Wii can support
 - how are fd value given and does it have a separate table for sockets